core/src/main/java/org/geysermc/geyser/util/Metrics.java

```diff
@@ -206,6 +208,11 @@ private static void sendData(ObjectNode data) throws Exception {
 
         // Compress the data to save bandwidth
         byte[] compressedData = compress(data.toString());
+        if (compressedData == null) {
+            // Something went wrong
+            connection.disconnect();
+            return;
+        }
```

Comment:
Instead, store and check if `data.toString()` is null before the connection is created.

Reply:
Yeah; was caused by the `compress()` method which could return null for a null input; which we already check and return at the very top